Test Failure: PostgreSQLAWSTestServlet.testCommonDataSourceIAM lacks Java 2 security permissions

Test Failure: com.ibm.ws.jdbc.fat.postgresql.PostgreSQLAWSTest.PostgreSQLAWSTestServlet.testCommonDataSourceIAM

```
PostgreSQLAWSTestServlet.testCommonDataSourceIAM:junit.framework.AssertionFailedError: 2023-08-23-10:14:23:145 ERROR: Caught exception attempting to call test method testCommonDataSourceIAM on servlet jdbc.fat.postgresql.web.PostgreSQLAWSTestServlet
java.lang.ExceptionInInitializerError
    at java.base/java.lang.J9VMInternals.ensureError(J9VMInternals.java:222)
    at java.base/java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:211)
    at software.amazon.jdbc.ds.AwsWrapperDataSource.setCredentialPropertiesFromUrl(AwsWrapperDataSource.java:349)
    at software.amazon.jdbc.ds.AwsWrapperDataSource.getConnection(AwsWrapperDataSource.java:84)
    at com.ibm.ws.rsadapter.impl.DatabaseHelper.getConnectionFromDatasource(DatabaseHelper.java:1467)
    at com.ibm.ws.rsadapter.impl.WSManagedConnectionFactoryImpl$7.run(WSManagedConnectionFactoryImpl.java:1032)
    at com.ibm.ws.rsadapter.impl.WSManagedConnectionFactoryImpl$7.run(WSManagedConnectionFactoryImpl.java:1019)
    at java.base/java.security.AccessController.doPrivileged(AccessController.java:784)
    at java.base/java.security.AccessController.doPrivilegedWithCombiner(AccessController.java:840)
    at com.ibm.ws.rsadapter.impl.WSManagedConnectionFactoryImpl.getConnectionUsingDS(WSManagedConnectionFactoryImpl.java:1019)
    at com.ibm.ws.rsadapter.impl.WSManagedConnectionFactoryImpl.getConnection(WSManagedConnectionFactoryImpl.java:848)
    at com.ibm.ws.rsadapter.impl.WSManagedConnectionFactoryImpl.createManagedConnection(WSManagedConnectionFactoryImpl.java:698)
    at com.ibm.ejs.j2c.FreePool.createManagedConnectionWithMCWrapper(FreePool.java:1376)
    at com.ibm.ejs.j2c.FreePool.createOrWaitForConnection(FreePool.java:1250)
    at com.ibm.ejs.j2c.PoolManager.reserve(PoolManager.java:1471)
    at com.ibm.ejs.j2c.ConnectionManager.allocateMCWrapper(ConnectionManager.java:576)
    at com.ibm.ejs.j2c.ConnectionManager.allocateConnection(ConnectionManager.java:308)
    at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:142)
    at com.ibm.ws.rsadapter.jdbc.WSJdbcDataSource.getConnection(WSJdbcDataSource.java:116)
    at jdbc.fat.postgresql.web.PostgreSQLAWSTestServlet.testCommonDataSourceIAM(PostgreSQLAWSTestServlet.java:84)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at componenttest.app.FATServlet.doGet(FATServlet.java:73)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:686)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:791)
    at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1260)
    at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:748)
    at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:445)
    at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1361)
    at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:5079)
    at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:318)
    at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:1038)
    at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:283)
    at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1248)
    at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:470)
    at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:429)
    at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:569)
    at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:503)
    at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:363)
    at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:330)
    at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:169)
    at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:77)
    at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:516)
    at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:586)
    at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:970)
    at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1059)
    at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:247)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
    at java.base/java.lang.Thread.run(Thread.java:858)
Caused by: java.security.AccessControlException: Access denied ("java.util.PropertyPermission" "software.amazon.jdbc.Driver.url.encoding" "read")
    at java.base/java.security.AccessController.throwACE(AccessController.java:177)
    at java.base/java.security.AccessController.checkPermissionHelper(AccessController.java:239)
    at java.base/java.security.AccessController.checkPermission(AccessController.java:386)
    at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:416)
    at java.base/java.lang.SecurityManager.checkPropertyAccess(SecurityManager.java:1160)
    at java.base/java.lang.System.getProperty(System.java:582)
    at software.amazon.jdbc.util.StringUtils.<clinit>(StringUtils.java:33)

    at componenttest.topology.utils.FATServletClient.assertTestResponse(FATServletClient.java:103)
    at componenttest.topology.utils.FATServletClient.runTest(FATServletClient.java:88)
    at componenttest.custom.junit.runner.SyntheticServletTest.invokeExplosively(SyntheticServletTest.java:44)
    at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:204)
    at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:364)
    at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:178)
    at org.testcontainers.containers.FailureDetectingExternalResource$1.evaluate(FailureDetectingExternalResource.java:29)

Further analysis is available at https://libh-proxy1.fyre.ibm.com/cognitive/defectAnalysis.html?failureGroupId=failure_group-84cefcfe-bce8-4165-b1d4-e8721d07f5c7
```

--- a/dev/com.ibm.ws.jdbc_fat_postgresql/publish/servers/server-PostgreSQLAWSTest/server.xml
+++ b/dev/com.ibm.ws.jdbc_fat_postgresql/publish/servers/server-PostgreSQLAWSTest/server.xml
@@ -63,4 +63,6 @@
     				wrapperPlugins="driverMetaData" wrapperDriverName="PostgreSQL JDBC Driver" />
     </dataSource>
 
+    <javaPermission codeBase="${shared.resource.dir}/aws/*" className="java.security.AllPermission"/>
+
 </server>

